server/src/main/java/hu/flowacademy/epsilon/sport_event_organizer/service/MatchService.java

```diff
@@ -0,0 +1,261 @@
+                for (int j = 2; j < 4; j++) {
+                    Match match = new Match();
+                    match.setTeamA(cupTeams.get(1));
+                    match.setTeamB(cupTeams.get(j));
+                    match.setGroupName(groupName);
+                    matches.add(match);
+                }
+                Match match = new Match();
+                match.setTeamA(cupTeams.get(2));
+                match.setTeamB(cupTeams.get(3));
+                match.setGroupName(groupName);
+                matches.add(match);
+                for (int j = 0; j < 3; j++) {
+                    cupTeams.get(j).setGroupName(groupName);
+                }
+                cupTeams.remove(3);
+                cupTeams.remove(2);
+                cupTeams.remove(1);
+                cupTeams.remove(0);
+                groupName = "Group2";
+            }
+        }
+
+        //Pálya kiosztás
+        matches.forEach(match -> match.setCup(cup));
+        String courtName = "Court1";
+        for (int i = 0; i < matches.size(); i++) {
+            matches.get(i).setCourtName(courtName);
+            if ((cup.getCourtCounter() == 2 && courtName == "Court2") || (cup.getCourtCounter() == 3 && courtName == "Court3") || (cup.getCourtCounter() == 4 && courtName == "Court4")) {
+                courtName = "Court1";
+            } else if (cup.getCourtCounter() >= 2 && courtName == "Court1") {
+                courtName = "Court2";
+            } else if (cup.getCourtCounter() >= 3 && courtName == "Court2") {
+                courtName = "Court3";
+            } else if (cup.getCourtCounter() >= 4 && courtName == "Court3") {
+                courtName = "Court4";
+            }
+        }
+
+        matches.forEach(match -> matchRepository.save(match));
+
+
+        return matches;
+    }
+
+    public List<Match> generatequalifierMatches(String cupName) {
+        Cup cup = cupService.getByName(cupName);
+        List<Team> cupTeams = new ArrayList<Team>(cupService.getApprovedTeams(cupName));
+        ArrayList<Match> matches = new ArrayList<>(cup.getMatches());
+        ArrayList<Team> winners = new ArrayList();
+        ArrayList<Team> group1 = new ArrayList();
+        ArrayList<Team> group2 = new ArrayList();
+
+        cupTeams.forEach(team -> team.setWinnerCounter(0));
+        cupTeams.forEach(team -> team.setQualified(false));
+//győztesek kiszámítása
+        for (Match match : matches) {
+            if (match.getTeamAScore() > match.getTeamBScore()) {
+                match.getTeamA().setWinnerCounter(match.getTeamA().getWinnerCounter() + 1);
+            } else {
+                match.getTeamB().setWinnerCounter(match.getTeamB().getWinnerCounter() + 1);
+            }
+            match.getTeamA().setGoalDifference(match.getTeamAScore() - match.getTeamBScore());
+            match.getTeamB().setGoalDifference(match.getTeamBScore() - match.getTeamAScore());
+        }
+
+//ki hányszor nyert
+        for (Team team : cupTeams) {
```

Comment:
Ez lehetne egy stream is! [link](https://www.javacodegeeks.com/2015/11/java-8-streams-api-grouping-partitioning-stream.html)